spring-cloud-stream-app-maven-plugin/src/main/java/org/springframework/cloud/stream/app/plugin/SpringCloudStreamAppMojo.java

```diff
@@ -309,6 +322,11 @@ private void postProcessGeneratedProject(GeneratableApp value, String appArtifac
 						"info.app.version=" + "@project.version@" + "\n" +
 						"management.endpoints.web.exposure.include=health,info,bindings" + "\n" +
 						"spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration" + "\n";
+				if(this.additionalAppProperties != null && !this.additionalAppProperties.isEmpty()){
+					for(String property : this.additionalAppProperties) {
+						applicationPropertiesContents = applicationPropertiesContents.concat(String.format("%s\n", property));
```

Comment:
Maybe we should prevent overriding provided properties, e.g. filter any entries that match,  e.g., `matches("^spring.application.name\w*=")` with a warning message

Reply:
I'm not sure I quite follow.   With us adding the ability to add properties to the application.properties the only ones that will be overridden are :
16) @PropertySource annotations on your @Configuration classes.
17) Default properties (specified by setting SpringApplication.setDefaultProperties)  
And this would be a deliberate act of the engineer creating the generated application of the starter.   But please argue with me, I always want the features to be the best they can be :-)

Reply:
`<additionalProperties>spring.application.name=foo</additionalProperties>`

Reply:
Is this the only one you want us to check for?

Reply:
I was thinking about all the hard coded properties up to L324.

Reply:
gotcha.  That makes perfect sense.  Will add it!